elasticsearch/example/kustomization.yaml

```diff
@@ -11,10 +11,22 @@ commonAnnotations:
   "app.uw.systems/tier": "tier_4"
 
 patches:
+  # Scaling the nodes down to 1
+  - target:
+      kind: StatefulSet
+      name: elasticsearch-master
+    patch: |-
+      - op: replace
+        path: /spec/replicas
+        value: 1
```

Comment:
Any reason to not just include this in `scale-down.yaml`?

Reply:
If I comment out `commonAnnotations`:

```patch
diff --git a/elasticsearch/example/kustomization.yaml b/elasticsearch/example/kustomization.yaml
index e3476a7..33b1af4 100644
--- a/elasticsearch/example/kustomization.yaml
+++ b/elasticsearch/example/kustomization.yaml
@@ -6,19 +6,11 @@ resources:
   - ../manifests
 
 # OpsLevel annotations
-commonAnnotations:
-  "app.uw.systems/description": "some es instance"
-  "app.uw.systems/tier": "tier_4"
+#commonAnnotations:
+#  "app.uw.systems/description": "some es instance"
+#  "app.uw.systems/tier": "tier_4"
 
 patches:
-  # Scaling the nodes down to 1- change amount of replicas and hostnames of the nodes
-  - target:
-      kind: StatefulSet
-      name: elasticsearch-master
-    patch: |-
-      - op: replace
-        path: /spec/replicas
-        value: 1
   - target:
       kind: StatefulSet
       name: elasticsearch-master
```

Then using building an annotations are set to the literal string `"null"` (haven't we seen this before somewhere?), see last line (extra context just to be clear which object it is):

```console
$ kustomize build  | grep -m1 -B30 '"null"'
        fsGroup: 1001
        fsGroupChangePolicy: Always
        supplementalGroups: []
        sysctls: []
      serviceAccountName: elasticsearch-metrics
---
apiVersion: apps/v1
kind: StatefulSet
metadata:
  annotations:
    app.uw.systems/repos: https://github.com/utilitywarehouse/shared-kustomize-bases/tree/main/elasticsearch
  labels:
    app: master
    app.kubernetes.io/component: master
    app.kubernetes.io/instance: elasticsearch
    app.kubernetes.io/name: elasticsearch
    app.kubernetes.io/version: 8.11.4
    helm.sh/chart: elasticsearch-19.16.1
  name: elasticsearch-master
spec:
  podManagementPolicy: Parallel
  replicas: 3
  selector:
    matchLabels:
      app.kubernetes.io/component: master
      app.kubernetes.io/instance: elasticsearch
      app.kubernetes.io/name: elasticsearch
  serviceName: elasticsearch-master-hl
  template:
    metadata:
      annotations: "null"
```

Reply:
Good shout! [Fixed it ](https://github.com/utilitywarehouse/shared-kustomize-bases/pull/17/commits/d81e80fecb40773de19f8c30116911914dde1f66)

Reply:
Diff after rendering example in main and in this branch: 
```

440c440
<   replicas: 1
---
>   replicas: 3
464,465d463
<         - name: ELASTICSEARCH_CLUSTER_MASTER_HOSTS
<           value: elasticsearch-master-0
487a486,487
>         - name: ELASTICSEARCH_CLUSTER_MASTER_HOSTS
>           value: elasticsearch-master-0 elasticsearch-master-1 elasticsearch-master-2
552c552
```

Reply:
> Diff after rendering example on main branch and on this branch

:thinking: it would be nice to have this diff output auto-generated from some CI (idea for the future)

Reply:
> 🤔 it would be nice to have this diff output auto-generated from some CI (idea for the future)
100% agree, this repo doesn't have any CICD for now. Nobody is using it yet, though, so I guess that's not where the priorities are :D

Reply:
[Created ticket](https://linear.app/utilitywarehouse/issue/DENA-433/shared-kustomize-bases-cicd)